Classes/Emogrifier.php

```diff
@@ -1741,27 +1751,43 @@ private function getNodesToExclude(\DOMXPath $xPath)
      */
     public function handleXpathError($type, $message, $file, $line, array $context)
     {
-        // Do not tie this conditional to debug mode as it causes unnecessary output. See further explanation in
-        // ::process() where this exception is caught.
-        if ($type === E_WARNING && isset($context['cssRule']['selector'])) {
-            throw new \InvalidArgumentException(
-                sprintf(
-                    '%1$s in selector >> %2$s << in %3$s on line %4$u',
-                    $message,
-                    $context['cssRule']['selector'],
-                    $file,
-                    $line
-                ),
-                1509279985
-            );
-        }
+        if ($type === E_WARNING) {
+            // Do not tie this conditional to debug mode as it causes unnecessary output. See further explanation in
+            // ::process() where this exception is caught.
+            if (isset($context['cssRule']['selector'])) {
+                throw new \InvalidArgumentException(
+                    sprintf(
+                        '%1$s in selector >> %2$s << in %3$s on line %4$u',
+                        $message,
+                        $context['cssRule']['selector'],
+                        $file,
+                        $line
+                    ),
+                    1509279985
+                );
+            }
 
-        // Catches eventual warnings generated by method getAllNodesWithStyleAttribute()
-        if ($type === E_WARNING && isset($context['xPath'])) {
-            throw new \RuntimeException(
-                sprintf('%1$s in %2$s on line %3$u', $message, $file, $line),
-                1509280067
-            );
+            // Catches eventual warnings generated by invalid selectors in method getNodesToExclude()
+            if (isset($context['selectorToExclude'])) {
+                throw new \InvalidArgumentException(
+                    sprintf(
+                        '%1$s in selector >> %2$s << in %3$s on line %4$u',
+                        $message,
+                        $context['selectorToExclude'],
+                        $file,
+                        $line
+                    ),
+                    1511683740
+                );
+            }
+
+            // Catches eventual warnings generated by method getAllNodesWithStyleAttribute()
+            if (isset($context['xPath'])) {
+                throw new \RuntimeException(
+                    sprintf('%1$s in %2$s on line %3$u', $message, $file, $line),
```

Comment:
Here, the some parameters to `sprintf` are missing.

Reply:
I am not sure what you mean here.